pkg/plugins/policies/meshhealthcheck/plugin/v1alpha1/plugin.go

```diff
@@ -158,6 +166,39 @@ func configure(
 	return nil
 }
 
+func applyToEgressRealResources(rs *core_xds.ResourceSet, proxy *core_xds.Proxy) error {
+	for _, resource := range proxy.ZoneEgressProxy.MeshResourcesList {
+		meshExternalServices := resource.Resources[meshexternalservice_api.MeshExternalServiceType]
```

Comment:
you can use `resource.ListOrEmpty(meshexternalservice_api.MeshExternalServiceType)`